lib/ansible/module_utils/facts/system/systemd.py

```diff
@@ -39,9 +39,17 @@ def collect(self, module=None, collected_facts=None):
             if rc != 0:
                 return systemd_facts
 
-            systemd_facts["systemd"] = {
-                "features": str(stdout.split("\n")[1]),
-                "version": int(stdout.split(" ")[1]),
-            }
+            systemd_version = str(stdout.split(" ")[1])
+
+            systemd_facts["systemd"] = {}
+            systemd_facts["systemd"]["features"] = str(stdout.split("\n")[1])
+            systemd_facts["systemd"]["full_version"] = str(
+                (stdout.split(" ")[2]).split(")")[0][1:],
+            )
```

Comment:
Same here: the second `str.split()` returns a list of strings, you take the first one and slice it from the second char into a new string.
```suggestion
            systemd_facts["systemd"]["full_version"] = (
                stdout.
                split(" ")[2].
                split(")")[0][1:]
            )
```